app/src/modules/theme/index.ts

```diff
@@ -0,0 +1,56 @@
+import { createSlice } from '@reduxjs/toolkit';
+import storage from 'utils/storage'
+
+export const themes = ['light' , 'dark'] as const
+export type Theme = typeof themes[number];
+
+interface ThemeState {
+  theme: Theme
+};
+
+const initialState: ThemeState = { theme: initialLoadTheme() };
+
+export const slice = createSlice({
+  name: 'selectedScreen',
+  initialState,
+  reducers: {
+    changeTheme: (state, action) => {
+      const theme = action.payload
+      storage.set('theme', theme)
+      setThemeAttribute(theme)
+      state.theme = theme
+    },
+  },
+});
+
+export const {
+  changeTheme
+} = slice.actions;
+
+export default slice.reducer;
+
+
+
+function initialLoadTheme(): Theme {
+  try {
+    const savedTheme = storage.get('theme')
+    const isOsDarkMode = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
+    const theme = (() => {
+      if(savedTheme && themes.includes(savedTheme)) return savedTheme
+      if(isOsDarkMode) return 'dark'
+      return 'light'
+    })() as Theme
+    setThemeAttribute(theme)
+    return theme
+  } catch {
+    // defaults to light theme if not available from OS or local storage
+    const theme = 'light'
+    setThemeAttribute(theme)
+    return theme
+  }
+}
+
+
+function setThemeAttribute(theme: Theme) {
+  document.documentElement.setAttribute("data-theme", theme);
+}
```

Comment:
Instead of having redux triggering this we can have in [Screen](https://github
.com/Rstankov/FocusedTask/blob/master/app/src/screens/index.tsx#L18) component:

```jsx
function useTheme() {
   const theme = useSelector(getTheme);
   React.useEffect(() => document.documentElement.setAttribute("data-theme", theme), [theme])
}
```

```jsx
export default function App() {
  useTheme();
  const selectedScreen = useSelector(getSelectedScreen);
  const Screen = (SCREENS as any)[selectedScreen] || SCREENS.task;

  return <Screen />;
}
```